packages/grafana-data/src/field/fieldDisplay.ts

```diff
@@ -298,3 +293,51 @@ export function getDisplayValueAlignmentFactors(values: FieldDisplay[]): Display
   }
   return info;
 }
+
+function createNoValuesFieldDisplay(options: GetFieldDisplayValuesOptions): FieldDisplay {
+  const defaultDisplayText = 'No data';
+  const { fieldOptions } = options;
+  const { defaults, override } = fieldOptions;
+
+  const config = getFieldProperties(defaults, {}, override);
+
+  if (!hasMappingForEmptyValue(config)) {
+    return {
+      name: defaultDisplayText,
+      field: {
+        ...defaults,
+      },
+      display: {
+        numeric: 0,
+        text: defaultDisplayText,
+      },
+    };
+  }
```

Comment:
want to try to find a way to get rid of this condition, feels wrong to have to check is this mapping exists

Reply:
Awesome that we think alike. I have already pushed an update to simplify it a bit.